lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_mme_handlers_with_injected_state.cpp

```diff
@@ -0,0 +1,166 @@
+    task_mme_app.detach();
+    task_sctp.detach();
+
+    s1ap_mme_init(&mme_config);
+
+    // add injection of state loaded in S1AP state manager
+    std::string magma_root = std::getenv("MAGMA_ROOT");
+    std::string state_data_path =
+        magma_root + "/" + DEFAULT_S1AP_STATE_DATA_PATH;
+    std::string data_folder_path =
+        magma_root + "/" + DEFAULT_S1AP_CONTEXT_DATA_PATH;
+    std::string data_list_path =
+        magma_root + "/" + DEFAULT_S1AP_CONTEXT_DATA_PATH + "data_list.txt";
+    assoc_id = 37;
+    stream_id = 0;
+    number_attached_ue = 2;
+
+    mock_read_s1ap_state_db(state_data_path);
+    name_of_ue_samples =
+        load_file_into_vector_of_line_content(data_folder_path, data_list_path);
+    mock_read_s1ap_ue_state_db(name_of_ue_samples);
+
+    state = S1apStateManager::getInstance().get_state(false);
+  }
+
+  virtual void TearDown() {
+    // Sleep to ensure that messages are received and contexts are released
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
+
+    send_terminate_message_fatal(&task_zmq_ctx_main_s1ap_with_injected_states);
+    destroy_task_context(&task_zmq_ctx_main_s1ap_with_injected_states);
+    itti_free_desc_threads();
+
+    free_mme_config(&mme_config);
+
+    // Sleep to ensure that messages are received and contexts are released
+    std::this_thread::sleep_for(std::chrono::milliseconds(200));
+  }
+
+ protected:
+  std::shared_ptr<MockMmeAppHandler> mme_app_handler;
+  std::shared_ptr<MockSctpHandler> sctp_handler;
+  s1ap_state_t* state;
+  sctp_assoc_id_t assoc_id;
+  sctp_stream_id_t stream_id;
+  std::vector<std::string> name_of_ue_samples;
+  int number_attached_ue;
+};
+
+TEST_F(S1apMmeHandlersWithInjectedStatesTest, GenerateUEContextReleaseCommand) {
+  ue_description_t ue_ref_p = {
+      .enb_ue_s1ap_id = 1,
+      .mme_ue_s1ap_id = 1,
+      .sctp_assoc_id = assoc_id,
+      .comp_s1ap_id = S1AP_GENERATE_COMP_S1AP_ID(assoc_id, 1)};
+
+  // State validation
+  ASSERT_TRUE(
+      is_enb_state_valid(state, assoc_id, S1AP_READY, number_attached_ue));
+  ASSERT_TRUE(is_num_enbs_valid(state, 1));
+
+  // Invalid S1 Cause returns error
+  ASSERT_EQ(RETURNerror, s1ap_mme_generate_ue_context_release_command(
+                             state, &ue_ref_p, S1AP_IMPLICIT_CONTEXT_RELEASE,
+                             INVALID_IMSI64, assoc_id, stream_id, 33, 10));
+  // Valid S1 Causes passess successfully
+  ASSERT_EQ(RETURNok, s1ap_mme_generate_ue_context_release_command(
+                          state, &ue_ref_p, S1AP_INITIAL_CONTEXT_SETUP_FAILED,
+                          INVALID_IMSI64, assoc_id, stream_id, 33, 10));
+
+  // State validation
+  ASSERT_TRUE(
+      is_enb_state_valid(state, assoc_id, S1AP_READY, number_attached_ue - 1));
+}
+
+TEST_F(S1apMmeHandlersWithInjectedStatesTest, HandleS1apPathSwitchRequest) {
+  ASSERT_EQ(task_zmq_ctx_main_s1ap_with_injected_states.ready, true);
+  ASSERT_TRUE(
```

Comment:
should we add asserts to make sure `number_attached_ue` is not changed before and after this test?